apps/chatbot/src/modules/vector_database.py

```diff
@@ -0,0 +1,173 @@
+import os
+import re
+import tqdm
+import logging
+
+from bs4 import BeautifulSoup, SoupStrainer
+from langchain_community.document_loaders import BSHTMLLoader
+
+from llama_index.core import Document, load_index_from_storage
+from llama_index.core.node_parser import HierarchicalNodeParser, get_leaf_nodes
+from llama_index.core import ServiceContext, VectorStoreIndex, StorageContext
+
+
+def filter_html_files(html_files):
+    pattern = re.compile(r"/v\d{1,2}.")
+    pattern2 = re.compile(r"/\d{1,2}.")
+    filtered_files = [file for file in html_files if not pattern.search(file) and not pattern2.search(file)]
+    return filtered_files
+
+
+def get_html_files(root_folder):
+    html_files = []
+    for root, dirs, files in os.walk(root_folder):
+        for file in files:
+            if file.endswith(".html"):
+                html_files.append(os.path.join(root, file))
+    return sorted(filter_html_files(html_files))
+
+
+def create_documentation(documentation_dir="./PagoPADevPortal/out/"):
+
+    if documentation_dir[-1] != "/":
+        documentation_dir += "/"
+
+    logging.info(f"Getting documentation from: {documentation_dir}")
+    
+    html_files = get_html_files(documentation_dir)
+    # parse_only = SoupStrainer("div", {"id": "page-content"})
+
+    documents = [Document(
+        text=(
+            "# Identità Chatbot\n\n\n\n"
```

Comment:
If you agree, I would consider moving the prompt in a dedicated file or settings file.